### PR TITLE
Some rendering fixes

### PR DESCRIPTION
**Pull request by scottwittenburg**

### CODE REVIEW

**Comment by jourdain** [comment]
Looks good to me but @martinken should look closer too...

File: Sources/Common/Core/ScalarsToColors/index.js [hunk 1]
```diff
@@ -125,25 +125,26 @@ function vtkScalarsToColors(publicAPI, model) {
...
     for (let i = component; i < size; i += tuple) {
-      newColors[(count * 4)] = convtFun(values[i]);
-      newColors[(count * 4) + 1] = convtFun(values[i + 1]);
-      newColors[(count * 4) + 2] = convtFun(values[i + 2]);
-      newColors[(count * 4) + 3] = a;
+      newValues[(count * 4)] = convtFun(values[i]);
+      newValues[(count * 4) + 1] = convtFun(values[i + 1]);
+      newValues[(count * 4) + 2] = convtFun(values[i + 2]);
+      newValues[(count * 4) + 3] = a;
       count++;
```
Comment (martinken) [inline Sources/Common/Core/ScalarsToColors/index.js]: Why operate on newValues as opposed to newColors? The method above is still using newColors so I am unsure what is behind the change.

**Comment by martinken** [comment]
+2

File: Sources/Common/Core/ScalarsToColors/index.js [hunk 2]
```diff
@@ -125,25 +125,26 @@ function vtkScalarsToColors(publicAPI, model) {
       newColors[count] = l;
       newColors[count + 1] = l;
       newColors[count + 2] = l;
-      newColors[count + 3] = convtFun(values[i + 1]);
+      newColors[count + 3] = convtFun(values[i + 1]) * alpha;
```
Comment (jourdain) [inline Sources/Common/Core/ScalarsToColors/index.js]: @martinken do you agree with the logic of Alpha multiplication here? It seems that it was done for all other use case except this one. So we were wondering if it was missing or not.

Comment (martinken) [inline Sources/Common/Core/ScalarsToColors/index.js]: I think it is correct. The code in VTK is
      outPtr[3] = static_cast<unsigned char>(a*alpha + 0.5);
which has the multiplication

File: Sources/Common/Core/ScalarsToColors/index.js (hunk 1 above)
Comment (scottwittenburg) [inline Sources/Common/Core/ScalarsToColors/index.js]: Good catch, I will fix the rest of those methods and push again.

**Pull request merged**